src/BuildScriptGenerator/DotNetCore/SdkVersionInfo.cs

```diff
@@ -0,0 +1,139 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System;
+
+namespace Microsoft.Oryx.BuildScriptGenerator.DotNetCore
+{
+    public class SdkVersionInfo : IComparable<SdkVersionInfo>, IEquatable<SdkVersionInfo>
+    {
+        public int Major { get; set; }
+
+        public int Minor { get; set; }
+
+        public int Feature { get; set; }
+
+        public int Patch { get; set; }
+
+        public int FeaturePatch { get; set; }
+
+        public string RawString { get; set; }
+
+        public bool IsPrerelease { get; set; }
+
+        public string PreviewVersion { get; set; }
+
+        public static SdkVersionInfo Parse(string sdkVersion)
+        {
+            TryParse(sdkVersion, out var result);
+            return result;
+        }
+
+        public static bool TryParse(string sdkVersion, out SdkVersionInfo result)
+        {
+            result = null;
+
+            var originalString = sdkVersion;
+
+            // Example: 3.1.100-preview1
+            var isPrerelease = false;
+            string previewVersion = null;
+            var index = sdkVersion.IndexOf("-preview", StringComparison.OrdinalIgnoreCase);
+            if (index >= 0)
+            {
+                isPrerelease = true;
+                previewVersion = sdkVersion.Substring(index + 1, sdkVersion.Length - (index + 1));
+                sdkVersion = sdkVersion.Substring(0, index);
+            }
+
+            // Example: 3.1.102
+            var sdkVersionParts = sdkVersion.Split(".");
```

Comment:
Any other way to parse string replacing Substring, indexOf and Split method?

Reply:
In general it is discouraged to use regex, so avoiding it here.